grails-app/controllers/org/pih/warehouse/order/OrderController.groovy

```diff
@@ -1096,6 +1096,66 @@ class OrderController {
     }
 
     def orderSummary = {
+        Order order = Order.get(params.id)
+        def orderItems = order?.orderItems
+        render(template: "orderSummary", model: [
+            isPurchaseOrder: order.isPurchaseOrder,
```

Comment:
@jmiranda I started moving things to the controller that are needed by the template and since there are used properties from order and order items I decided to leave it as is -> fetch order and pull order items (and other required data) from it instead of fetching only order items.

Reply:
Ok. I think we can refactor this a bit more when we move to Grails 3. As @kchelstowski mentioned this might look better as queries defined in the data service. 

This is the one change I would like to see so we can move this in that direction when the time comes
```
def orderSummary = { 
  Map orderSummary = orderService.getOrderSummary(String id)
  render render(template: "orderSummary", model: orderSummary)
}
```
with the order summary method looking something like this. And since we know that we're just using this method for reading data we can add in some extra savings. 
```
@Transactional(readOnly=true)
Map getOrderSummary(String id) { 
  Order order = Order.read(params.id)
    return [
      isPurchaseOrder: order.isPurchaseOrder
      isPutawayOrder: order.isPutawayOrder,
      orderItems: orderItems,
      anySupplierCode: orderItems?.any { it.productSupplier?.supplierCode },
      anyManufacturerName: orderItems?.any { it.productSupplier?.manufacturerName },
      anyManufacturerCode: orderItems?.any { it.productSupplier?.manufacturerCode },
      currencyCode: order.currencyCode ?: grailsApplication.config.openboxes.locale.defaultCurrencyCode,
      subtotal: order.subtotal,
      totalAdjustments: order.totalAdjustments,
      total: order.total
    ]
}
```
This gives us a clean / DRY way of retrieving the data that can be refactored in the future. We'll also have an easier time unit testing and performance tuning in the future since all of the data retrieval is centralized in a single service method.

This is one step away from @kchelstowski's command object approach. And I don't see us needing that because the map works well enough as a DTO. The command object would be necessary if we then needed to add more methods and validation to the object.